src/sagemaker_huggingface_inference_toolkit/transformers_utils.py

```diff
@@ -273,8 +283,24 @@ def get_pipeline(task: str, device: int, model_dir: Path, **kwargs) -> Pipeline:
     else:
         kwargs["tokenizer"] = model_dir
 
-    # load pipeline
-    hf_pipeline = pipeline(task=task, model=model_dir, device=device, **kwargs)
+    if TRUST_REMOTE_CODE and os.environ.get("HF_MODEL_ID", None) is not None and device == 0:
+        torch_dtype = torch.bfloat16 if torch.cuda.get_device_capability()[0] == 8 else torch.float16
+        tokenizer = AutoTokenizer.from_pretrained(os.environ["HF_MODEL_ID"])
+
+        hf_pipeline = pipeline(
+            task=task,
+            model=os.environ["HF_MODEL_ID"],
+            tokenizer=tokenizer,
+            trust_remote_code=TRUST_REMOTE_CODE,
+            model_kwargs={"device_map": "auto", "torch_dtype": torch_dtype},
+        )
+    elif is_diffusers_available() and task == "text-to-image":
```

Comment:
Would it makes sense to go to `else` case if `task == "text-to-image" but `is_diffusers_available` returns `False`?

Reply:
no. There you need both.

Reply:
You mean we should rather error? That when `task-to-image` == True but diffusers not available?

Reply:
Yeah, probably. Right now, the code does something that doesn't make sense in that case, because it falls through to the `else` clause.

I said that from the beginning. But yeah, at this point this back and forth is a bit ridiculous, so whatever. It's probably fine in practice. But still, this if-elif-else logic seems kind of wrong. I don't know how to be more clear.